xarray/test/test_dataarray.py

```diff
@@ -1407,6 +1407,42 @@ def test_resample_first(self):
                              name='time')
         self.assertDataArrayIdentical(expected, actual)
 
+    def test_resample_first_keep_attrs(self):
+        times = pd.date_range('2000-01-01', freq='6H', periods=10)
+        array = DataArray(np.arange(10), [('time', times)])
+        array.attrs['meta'] = 'data'
+
+        resampled_array = array.resample('1D', dim='time', how='first', keep_attrs=True)
```

Comment:
I get what you are saying but I disagree with the premise. Remember, the reason you are writing this test in the first place was that  `how = {first, last}` were behaving differently. So the point here should be to make sure that they don't do that anymore. It is because they are treated differently that we should test them together. The way I see it is that this is a single case.

In practice, we make multiple assertions in single unit test all the time (browse the xarray code base for plenty of examples). If the first assertion fails in a test, it doesn't continue and that isn't a problem because the test failed.

Lastly, no, we'll never get an exhaustive list but that shouldn't stop us from thinking about the known edge cases. The known edge case in question is that `how = {first, mean}` treat the attributes differently.  If we are concerned about the `resample.reduce` treatment of attributes, maybe add a `how=np.mean` along with the others.